### PR TITLE
Light Refactor and Clean-up AWS Provider

### PR DESCRIPTION
**Pull request by vincentkoc**
Cleaning up and some light Python refactoring of the AWS provider.

- Use of f-string for concat
- UTC using true tz
- Missing type hints
- pathlib for file handler

### CODE REVIEW

File: tests/providers/amazon/aws/log/test_s3_task_handler.py [hunk 1]
```diff
@@ -75,10 +76,8 @@ def setUp(self):
 
     def tearDown(self):
         if self.s3_task_handler.handler:
-            try:
+            with contextlib.suppress(Exception):
```
Comment (ferruzzi) [inline tests/providers/amazon/aws/log/test_s3_task_handler.py:79]: Oh.  That's a new one to me.  Nice.

File: airflow/providers/amazon/aws/hooks/glue.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def __init__(
                 raise ValueError("Cannot specify num_of_dpus with custom WorkerType")
         elif not worker_type_exists and num_workers_exists:
             raise ValueError("Need to specify custom WorkerType when specifying NumberOfWorkers")
-        elif worker_type_exists and not num_workers_exists:
+        elif worker_type_exists:
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/hooks/glue.py]: I might argue that while these are functionally the same, the old way was a little more clear about the intent.  Just a thought.

**Review by ferruzzi — APPROVED** [review]
A lot of work went into this.  Thanks.   Left one small nitpick suggestion.

**Review by vincbeck — APPROVED** [review]

File: tests/providers/amazon/aws/utils/eks_test_utils.py [hunk 3]
```diff
@@ -164,10 +164,14 @@ def region_matches_partition(region: str, partition: str) -> bool:
...
-            return partition == expected_partition
-    return partition == "aws"
+    return next(
+        (
+            partition == expected_partition
+            for prefix, expected_partition in valid_matches
+            if region.startswith(prefix)
+        ),
+        partition == "aws",
+    )
```
Comment (o-nikolas) [inline tests/providers/amazon/aws/utils/eks_test_utils.py]: Nit: Defining a generator inline and calling next with the default is fun Python but I think it is a bit unnecessary, it's turning 4 lines of very readable code into 8 lines of obtuse Python.

**Review by o-nikolas — APPROVED** [review]
Thanks for the contribution! I just left one nit otherwise good stuff :+1:

**Comment by koconder** [comment]
Thanks @o-nikolas @ferruzzi @vincbeck i have made the required changes

**Comment by koconder** [comment]
Rebased branch as some tests failed which previously passed

**Comment by ferruzzi** [comment]
I suspect most of the committers are busy with Airflow Summit this week, but hopefully we can find someone to get this merged for you next week.

**Comment by koconder** [comment]
@ferruzzi yea, no rush all good :)

**Review by mik-laj — APPROVED** [review]

**Pull request merged**